### PR TITLE
New Route 'Create' for Booksuggestion

### PR DESCRIPTION
**Pull request by mcolombo87**
### Summary
New Route 'Create' for Booksuggestion. 
### ScreenShoots
Without user:
image
With user:
image
### Trello
https://trello.com/c/YxpHFHqx/27-crear-ruta-para-creaci%C3%B3n-de-un-booksuggestion
### Notes
Are not need to be authenticated for make a suggestion, if you are then Book Suggestion will have your user_id.

### CODE REVIEW

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module Api
+  module V1
+    class BookSuggestionsController < ApplicationController
+      def create
+        params[:book_suggestion][:user_id] = current_user.id if user_signed_in?
+        new_book_sugges = BookSuggestion.new(book_sugges_params)
```
Comment (facucompe) [inline app/controllers/api/v1/book_suggestions_controller.rb]: `BookSuggestion.new(book_sugges_params, user: current_user)`

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+module Api
+  module V1
+    class BookSuggestionsController < ApplicationController
+      def create
+        params[:book_suggestion][:user_id] = current_user.id if user_signed_in?
```
Comment (facucompe) [inline app/controllers/api/v1/book_suggestions_controller.rb]: Remove this line. For this endpoint you need to be authenticated, add a `before_action`

**Review by facucompe — CHANGES_REQUESTED** [review]

**Review by floresv — COMMENTED** [review]
I agree with Facu, please fix it.

File: app/controllers/api/v1/book_suggestions_controller.rb (hunk 2 above)
Comment (mcolombo87) [reply app/controllers/api/v1/book_suggestions_controller.rb]: According with the Trello's Card its not necessary to be authenticated. He says "Wolox Training QUIERE que se implemente el CREATE de un BookSuggestion sin autenticación para que un usuario invitado en la plataforma pueda crear una sugerencia." So, I made that for, if the user is authenticated save his id in the BookSuggest. In the card before it, he request me that BookSuggestion can save an User Reference. This way I'm accomplish with both cards are requesting me.

**Comment by mcolombo87** [comment]
I'd responded to the review with my reasons for do it that.

File: app/controllers/api/v1/book_suggestions_controller.rb (hunk 2 above)
Comment (mcolombo87) [reply app/controllers/api/v1/book_suggestions_controller.rb]: Done, I changed it. Now, only authenticated users can send a Book Suggestion.

File: app/controllers/api/v1/book_suggestions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+module Api
+  module V1
+    class BookSuggestionsController < ApplicationController
+      before_action :authenticate_user!
+      def create
+        new_book_sugges = BookSuggestion.new(book_sugges_params, user: current_user)
+        if new_book_sugges.save
+          render json: new_book_sugges, serializer: BookSuggestionSerializer, status: :created
```
Comment (floresv) [inline app/controllers/api/v1/book_suggestions_controller.rb:8]: Maybe a `return reder if ...` save some code lines

Comment (floresv) [reply app/controllers/api/v1/book_suggestions_controller.rb:8]: Another thing first errors, at the end good response. Read https://github.com/rubocop-hq/ruby-style-guide

**Review by floresv — COMMENTED** [review]
Wait for facu approves

**Review by facucompe — APPROVED** [review]
LGTM, change what Javi said if you want

**Pull request merged**